### DISCUSSION TITLE
support next.js `dynamic` component

### DISCUSSION THREAD

**Discussion by y-nk** [remix-run/react-router#10670] [Proposals]
I'd just think it'd be nice if we could eventually use `next/dynamic` into the route definition array. For now, the type aren't matching so we can't do:

```ts
const routes = [{
  path: '/',
  element: dynamic(() => import('~/pages/Home'), { ssr: false })
}]
```

the way to work around it is:

```tsx
const Home = dynamic(() => import('~/pages/Home'), { ssr: false })

const routes = [{
  path: '/',
  element: <Home />
}]
```

which i think is a bit cumbersome.

PS: i also asked Next to support a component mode for their `dynamic` which at the moment is not permitted → https://github.com/vercel/next.js/discussions/52236

**Comment by ivan-kleshnin** [comment]
Have you tried native `React.lazy` https://react.dev/reference/react/lazy ? Has to be combined with `remix-utils/client-only` to move it outside SSR.

```ts
import {lazy} from "react"
import {ClientOnly} from "remix-utils/client-only"

export let ApexChart = lazy(() => import("./ApexChart"))

export function YourComponent() {
  return <ClientOnly>
    {() => <ApexChart/>}
  </ClientOnly>
}
```
